mxcubecore/HardwareObjects/MotorsNPosition.py

```diff
@@ -182,6 +183,10 @@ def motor_value_changed(self, position=None):
         self.update_multi_value()
 
     def update_multi_value(self):
+        if self._updating_multi_value:
+            pass
+        self._updating_multi_value = True
```

Comment:
This does not make sense to me. As far as I can tell this `if` clause does not do anything. Maybe the `if` clause needs to be removed entirely.

```suggestion
        self._updating_multi_value = True
```

Reply:
The change looks problematic too. The basic ideas seems to be to avoid having this function called indirectly while it is already being executed. To achieve this
1) decide what shoudl be returned from these recursive calls. Since the 'no current position' value seems to be -1, maybe that would be a better choice. 

2) I would propose to start with

  current_idx = -1
    if not self._updating_multi_value:
      self._updating_multi_value = True
      try:
        Put the entire body of the function here
        Then end with:

      finally:
    self._updating_multi_value = False
  return current_idx

Reply:
Thanks for the proposal. I've updated it now.

Reply:
@fabcor-maxiv , I've updated it according to the latest suggestions, but additionally the corresponding hardware objects are rewritten and using different class. And it also do not produce any reccurency errors. Ive tested it, and it seems to be happy, but I would keep this class as an emergency rollback solution if this is ok.